analyzers/src/SonarAnalyzer.CSharp/Rules/BlazorQueryParameterTypeShouldBeSupported.cs

```diff
@@ -0,0 +1,106 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class BlazorQueryParameterTypeShouldBeSupported : SonarDiagnosticAnalyzer
+{
+    private const string DiagnosticId = "S6797";
+    private const string MessageFormat = "Query parameter type '{0}' is not supported.";
+
+    private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
+
+    private static readonly ISet<KnownType> SupportedTypes = new HashSet<KnownType>
+    {
+        KnownType.System_Boolean,
+        KnownType.System_DateTime,
+        KnownType.System_Decimal,
+        KnownType.System_Double,
+        KnownType.System_Single,
+        KnownType.System_Int32,
+        KnownType.System_Int64,
+        KnownType.System_String,
+        KnownType.System_Guid
+    };
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterCompilationStartAction(cc =>
+        {
+            if (cc.Compilation.GetTypeByMetadataName(KnownType.Microsoft_AspNetCore_Components_RouteAttribute) is null)
+            {
+                return;
+            }
+
+            cc.RegisterSymbolAction(c =>
```

Comment:
Educational: what is the advantage of registering for a symbol instead of a `PropertyDeclarationSyntax` node for example?

Reply:
I need to check the route, I could do this through a ClassDeclarationSyntax, but I will need in any case to get the class Symbol to retrieve all properties that can be across multiple files.

Not sure though it is better than using `RegisterSyntaxNodeAction`